airbyte-cdk/java/airbyte-cdk/s3-destinations/src/testFixtures/kotlin/io/airbyte/cdk/integrations/destination/s3/S3DestinationAcceptanceTest.kt

```diff
@@ -185,6 +203,304 @@ protected constructor(protected val outputFormat: FileUploadFormat) :
+            )
+        val configuredCatalog = CatalogHelpers.toDefaultConfiguredCatalog(catalog)
+        configuredCatalog.streams.forEach {
+            it.withSyncMode(syncMode)
+                .withDestinationSyncMode(destinationSyncMode)
+                .withSyncId(syncId)
+                .withGenerationId(generationId)
+                .withMinimumGenerationId(minimumGenerationId)
+        }
+        return Pair(configuredCatalog, catalog)
+    }
+
+    private fun getFirstSyncMessagesFixture1(
+        configuredCatalog: ConfiguredAirbyteCatalog,
+        streamStatus: AirbyteStreamStatusTraceMessage.AirbyteStreamStatus
+    ): List<AirbyteMessage> {
+        val descriptor = StreamDescriptor().withName(configuredCatalog.streams[0].stream.name)
+        return listOf(
+            AirbyteMessage()
+                .withType(AirbyteMessage.Type.RECORD)
+                .withRecord(
+                    AirbyteRecordMessage()
+                        .withStream(configuredCatalog.streams[0].stream.name)
+                        .withEmittedAt(Instant.now().toEpochMilli())
+                        .withData(
+                            Jsons.jsonNode(
+                                ImmutableMap.builder<Any, Any>()
+                                    .put("id", 1)
+                                    .put("currency", "USD")
+                                    .put("date", "2020-03-31T00:00:00Z")
+                                    .put("HKD", 10.1)
+                                    .put("NZD", 700.1)
+                                    .build(),
+                            ),
+                        ),
+                ),
+            AirbyteMessage()
+                .withType(AirbyteMessage.Type.STATE)
+                .withState(
+                    AirbyteStateMessage()
+                        .withData(Jsons.jsonNode(ImmutableMap.of("checkpoint", 2))),
+                ),
+            AirbyteMessage()
+                .withType(AirbyteMessage.Type.TRACE)
+                .withTrace(
+                    AirbyteTraceMessage()
+                        .withType(AirbyteTraceMessage.Type.STREAM_STATUS)
+                        .withStreamStatus(
+                            AirbyteStreamStatusTraceMessage()
+                                .withStreamDescriptor(descriptor)
+                                .withStatus(streamStatus)
+                        ),
+                ),
+        )
+    }
+
+    private fun getSyncMessagesFixture2(): List<AirbyteMessage> {
+        return MoreResources.readResource(
+                DataArgumentsProvider.Companion.EXCHANGE_RATE_CONFIG.getMessageFileVersion(
+                    getProtocolVersion(),
+                ),
+            )
+            .trim()
+            .lines()
+            .map { Jsons.deserialize(it, AirbyteMessage::class.java) }
+    }
+
+    @Test
+    fun testOverwriteSyncPreRefreshAndPostSupport() {
+        if (!implementsOverwrite()) {
```

Comment:
you can replace this with
```kotlin
assumeTrue(implementsOverride(), "Destination's spec.json does not support overwrite sync mode.")
```

and it will mark the test as skipped instead of successful.

Reply:
copy-pasta.  will change for all new tests.